sklearn/feature_selection/_univariate_selection.py

```diff
@@ -659,11 +656,17 @@ class SelectKBest(_BaseFilter):
     (1797, 20)
     """
 
+    _parameter_constraints = {
+        **_BaseFilter._parameter_constraints,  # type: ignore
+        "k": [StrOptions({"all"}), Interval(Integral, -1, None, closed="neither")],
+    }
+
     def __init__(self, score_func=f_classif, *, k=10):
         super().__init__(score_func=score_func)
         self.k = k
 
     def _check_params(self, X, y):
+        super()._check_params(X, y)
         if not (self.k == "all" or 0 <= self.k <= X.shape[1]):
             raise ValueError(
                 "k should be >=0, <= n_features = %d; got %r. "
```

Comment:
Could you change this check with an f-string since we change the validation.

Reply:
I realized that the check can be simplified a little further since the only thing we need to check now is for `k > n_features` (assuming it is a numeric value and not `'all'`). I converted the error message to an f-string as is, but I'm not sure if that is what you were looking for exactly.

Reply:
Oh yes, that's true. We can only check for the numerical part now. I would be in favour of doing this. Good catch.